packages/@aws-cdk/aws-secretsmanager/test/secret.test.ts

```diff
@@ -242,6 +242,60 @@ test('grantRead', () => {
   });
 });
 
+test('grantRead cross account', () => {
+  // GIVEN
+  const key = new kms.Key(stack, 'KMS');
+  const secret = new secretsmanager.Secret(stack, 'Secret', { encryptionKey: key });
+  const principal = new iam.AccountPrincipal('1234');
+
+  // WHEN
+  secret.grantRead(principal, ['FOO', 'bar']).assertSuccess();
+
+  // THEN
+  expect(stack).toHaveResource('AWS::SecretsManager::ResourcePolicy', {
```

Comment:
Can you validate the KMS policy changes as well in this test?

Reply:
Good idea. Added.

The permissions are the same as described in https://aws.amazon.com/premiumsupport/knowledge-center/secrets-manager-share-between-accounts/ with exception of `kms:EncryptionContext:SecretARN` condition key because it 1) creates a chicken-egg problem between KMS and SecretsManager if the Secret is automatically named by CFN/CDK. 2) it's redundant since permissions are already controlled by SecretsManager.